idea-plugin/src/com/reason/bs/BscProcessListener.java

```diff
@@ -0,0 +1,48 @@
+package com.reason.bs;
+
+import com.intellij.execution.process.ProcessEvent;
+import com.intellij.execution.process.ProcessListener;
+import com.intellij.openapi.util.Key;
+import com.reason.Log;
+import com.reason.ide.annotations.OutputInfo;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.List;
+
+public class BscProcessListener implements ProcessListener {
+
+    private static final Log LOG = Log.create("bsc.process");
+
+    final BsLineProcessor m_lineProcessor = new BsLineProcessor(LOG);
+    final StringBuilder m_builder = new StringBuilder();
+
+    @Override
+    public void startNotified(@NotNull ProcessEvent event) {
+        LOG.trace("start " + event);
+    }
+
+    @Override
+    public void processTerminated(@NotNull ProcessEvent event) {}
+
+    @Override
+    public void processWillTerminate(@NotNull ProcessEvent event, boolean willBeDestroyed) {}
+
+    @Override
+    public void onTextAvailable(@NotNull ProcessEvent event, @NotNull Key outputType) {
+        String text = event.getText();
+        if (text != null) {
+            m_builder.append(text);
+            if (text.endsWith("\n")) {
+                m_lineProcessor.onRawTextAvailable(m_builder.toString());
+                if (LOG.isTraceEnabled()) {
+                    LOG.trace(m_builder.toString().replace("\n", ""));
+                }
+                m_builder.setLength(0);
```

Comment:
I don't understand why you are using a StringBuilder instead of using text variable directly

Reply:
this was copy-pasted. onTextAvailable is called frequently but we only want to trigger our processor when we see a newline.

`StringBuilder` pre-allocates memory for storing a string whereas string concatenation reallocates memory on each concatenation. Think resizing an ArrayList vs resizing an Array[]

Reply:
I didn't read code correctly, you're right